server/app/api/v1/endpoints/subscriptions.py

```
@@ -0,0 +1,2078 @@
+            meta = dict(subscription.subscription_metadata or {})
+            rooftop_ids = meta.get("rooftop_ids") or []
+            if not isinstance(rooftop_ids, list):
+                rooftop_ids = []
+            rid = str(request.rooftop_id)
+            if rid not in {str(x) for x in rooftop_ids}:
+                meta["rooftop_ids"] = [*rooftop_ids, rid]
+                subscription.subscription_metadata = meta
+                flag_modified(subscription, "subscription_metadata")
+                await db.commit()
+                await db.refresh(subscription)
+        except Exception as e:
```

Ruff (0.14.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)